### ISSUE TITLE
Allow KNNIndex classes to take **kwargs arguments and pass them to KNN library

### ISSUE THREAD

**Issue by dkobak** [pavlin-policar/openTSNE#256]
See title. We discussed it at Dagstuhl. Should be useful for PyNNdescent, but could potentially be useful for Annoy/HNSW/etc at well.

We could potentially allow Affinity classes to take **kwargs as well and pass them to the KNNIndex. Here I'm less sure if that's a good design idea.

**Comment by pavlin-policar** [comment]
Yes, we definitely need to add that! IMO, a general **kwargs would be messy, and something like like a `knn_kwargs` argument would be better. Scikit-learn also does something similar in [their KNN classes](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NearestNeighbors.html), but they call it "params". I prefer "kwargs". What do you think?
